docs/jdbc.md

```diff
@@ -0,0 +1,456 @@
+<dependency>
+    <groupId>com.google.cloud.sql</groupId>
+    <artifactId>mysql-socket-factory-connector-j-8</artifactId>
+    <version>1.14.1</version>
+</dependency>
+```
+
+Gradle
+
+```gradle
+compile 'com.google.cloud.sql:mysql-socket-factory-connector-j-8:1.14.1'
+```
+
+<!-- {x-version-update-end} -->
+
+##### Maria DB
+
+<!-- {x-version-update-start:mariadb-socket-factory:released} -->
+Maven
+
+```maven-pom
+<dependency>
+    <groupId>com.google.cloud.sql</groupId>
+    <artifactId>mariadb-socket-factory</artifactId>
+    <version>1.14.1</version>
+</dependency>
+```
+
+Gradle
+
+```gradle
+compile 'com.google.cloud.sql:mariadb-socket-factory:1.14.1'
+```
+
+**Note:** Also include the JDBC Driver for
+MariaDB, `org.mariadb.jdbc:mariadb-java-client:<LATEST-VERSION>`
+<!-- {x-version-update-end} -->
+
+##### Postgres
+
+<!-- {x-version-update-start:postgres-socket-factory:released} -->
+Maven
+
+```maven-pom
+<dependency>
+    <groupId>com.google.cloud.sql</groupId>
+    <artifactId>postgres-socket-factory</artifactId>
+    <version>1.14.1</version>
+</dependency>
+```
+
+Gradle
+
+```gradle
+compile 'com.google.cloud.sql:postgres-socket-factory:1.14.1'
+```
+
+**Note:**  Also include the JDBC Driver for
+PostgreSQL, `org.postgresql:postgresql:<LATEST-VERSION>`
+<!-- {x-version-update-end} -->
+
+##### SQL Server
+
+<!-- {x-version-update-start:cloud-sql-connector-jdbc-sqlserver:released} -->
+Maven
+
+```maven-pom
+<dependency>
+    <groupId>com.google.cloud.sql</groupId>
+    <artifactId>postgres-socket-factory</artifactId>
+    <version>1.14.1</version>
+</dependency>
+```
+
+Gradle
+
+```gradle
+compile 'com.google.cloud.sql:cloud-sql-connector-jdbc-sqlserver:1.14.1'
+```
+
+**Note:**  Also include the JDBC Driver for SQL
+Server, `com.microsoft.sqlserver:mssql-jdbc:<LATEST-VERSION>`.
+<!-- {x-version-update-end} -->
+<!-- {x-release-please-end} -->
+
+### Creating the JDBC URL
+
+When specifying the JDBC connection URL, add the additional parameters:
+
+| Property         | Value                                                             |
+|------------------|-------------------------------------------------------------------|
+| socketFactory    | <SOCKET_FACTORY_CLASS>                                            |
+| cloudSqlInstance | The instance connection name (found on the instance details page) |
+| user             | Database username                                                 |
+| password         | Database user's password                                          |
+
+Replace <SOCKET_FACTORY_CLASS> with the class name specific to your database.
+
+#### Mysql
+
+Base JDBC URL: `jdbc:mysql:///<DATABASE_NAME>`
+
+SOCKET_FACTORY_CLASS: `com.google.cloud.sql.mysql.SocketFactory`
+
+The full JDBC URL should look like this:
+
+```
+jdbc:mysql:///<DATABASE_NAME>?cloudSqlInstance=<INSTANCE_CONNECTION_NAME>&socketFactory=com.google.cloud.sql.mysql.SocketFactory&user=<MYSQL_USER_NAME>&password=<MYSQL_USER_PASSWORD>
```

Comment:
Should we break up these long strings like we did below? It would be best to pick one approach and be consistent. Both seem fine, although this is a long line generally.

Reply:
You're right, shorter lines are better.